examples/llm_compression/torch/qat_with_lora/main.py

```diff
@@ -0,0 +1,418 @@
+# Copyright (c) 2025 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import random
+import shutil
+import sys
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict, List
+
+import torch
+import torch.nn.functional as F
+from datasets import load_dataset
+from optimum.exporters.openvino.convert import export_from_model
+from optimum.intel.openvino import OVModelForCausalLM
+from torch import Tensor
+from torch import nn
+from torch.utils.tensorboard import SummaryWriter
+from tqdm import tqdm
+from tqdm import trange
+from transformers import AutoModelForCausalLM
+from transformers import AutoTokenizer
+from whowhatbench import TextEvaluator
+
+import nncf
+from nncf.data.dataset import Dataset
+from nncf.parameters import CompressionFormat
+from nncf.parameters import CompressWeightsMode
+from nncf.parameters import StripFormat
+from nncf.quantization.quantize_model import compress_weights
+from nncf.torch.model_creation import load_from_config
+from nncf.torch.quantization.layers import AsymmetricLoraQuantizer
+from nncf.torch.quantization.layers import SymmetricLoraQuantizer
+
+
+def get_wikitext2(nsamples: int, seqlen: int, tokenizer: Any, device: torch.device) -> List[Tensor]:
+    """
+    Loads and processes the Wikitext-2 dataset for training.
+
+    :param nsamples: Number of samples to generate.
+    :param seqlen: Sequence length for each sample.
+    :param tokenizer: Tokenizer to encode the text.
+    :param device: Device to move the tensors to (e.g., 'cpu' or 'cuda').
+    :return: A list of tensors containing the tokenized text samples.
+    """
+    traindata = load_dataset("wikitext", "wikitext-2-raw-v1", split="train")
+    limit = nsamples * seqlen // 4  # ~1k for 128 samples with seqlen=32 to be aligned with optimum
+    text = "".join([" \n" if s == "" else s for s in traindata["text"][:limit]])
+    trainenc = tokenizer(text, return_tensors="pt")
+    trainloader = []
+    for _ in range(nsamples):
+        # Crop a sequence of tokens of length seqlen starting at a random position
+        i = torch.randint(0, trainenc.input_ids.shape[1] - seqlen - 1, (1,)).item()
+        j = i + seqlen
+        inp = trainenc.input_ids[:, i:j].to(device)
+        trainloader.append(inp)
+    return trainloader
+
+
+def set_seed(seed) -> None:
```

Comment:
fyi, there is `set_seed` in `transformers` library.

Reply:
Thanks, reused